### ISSUE TITLE
Missing functionality for sizeof

### ISSUE THREAD

**Issue by madmann91** [AnyDSL/thorin#57]
The `sizeof` PrimOp has no argument, one type argument (the type for which we want to get the size), and one return type (i32). This is annoying, because the `import` transformation requires the `vrebuild` function to be of the form:
```cpp
const Def* SizeOf ::vrebuild(World& to, Defs, const Type* t)
```
Here, `t` is the _return type_ (i.e. i32), and we cannot rebuild `size_of->of()` by calling `vrebuild` on its type, because it is private.

The correct fix for this is to add a `Type2Type` argument to `vrebuild` for PrimOps as well. This is a bit of a change though, and I would like to get your approval first.

*Note*: This fixes the SEGFAULT issue in Stincilla for the infer branch. The last remaining issue will be the `reserve_shared` test.

**Comment by madmann91** [comment]
Another option is just to hack it and write:
```cpp
const Def* SizeOf ::vrebuild(World& to, Defs,     const Type* t) const { return to.size_of(of()->rebuild(to, {} /* Monomorphic only */), loc(), name); }
```
But we will not support polymorphic types then.

**Comment by leissa** [comment]
How about giving size_of a dummy argument? just to track the type. So the constructor for size_of would be:
```C++
world.size_of(world.bottom(whatever))
```
Then, rebuild would work.

**Comment by madmann91** [comment]
That's probably a better option. Let's do that. I then drop the `of_` member of SizeOf.

**Comment by madmann91** [comment]
Fixed in 8fb2180f62ad60cf61f67d75348166885ec2b635.
